cli/cmd/inject.go

```diff
@@ -100,6 +100,11 @@ sub-folders, or coming from stdin.`,
 		&manualOption, "manual", manualOption,
 		"Include the proxy sidecar container spec in the YAML output (the auto-injector won't pick it up, so config annotations aren't supported) (default false)",
 	)
+	flags.UintVar(
+		&options.waitBeforeExitSeconds, "wait-before-exit-seconds", options.waitBeforeExitSeconds,
+		"The periods during which the proxy sidecar container must stay alive while its pod is terminating. "+
+			"Must be smaller than terminationGracePeriodSeconds for the pod (default 0)",
+	)
```

Comment:
TL;DR notes for myself:
- move the logic under `flagSet()` in `root.go`
- rename the option to `proxy-wait-before-exit`
- update the desc: "The periods" > "The period"
- update the desc: remove redundant "container" word 

What about the "-seconds" suffix? Worth keeping it everywhere or should I remove it and rename all the variables?

Reply:
The move to `root.go` is no longer needed according to my comment below

Reply:
> What about the "-seconds" suffix? Worth keeping it everywhere or should I remove it and rename all the variables?

I would remove it from the `linkerd inject` command, and accept a `duration` type, to keep it in line with other options like `--identity-clock-skew-allowance` and `--identity-issuance-lifetime`. And then when the annotation gets interpreted in `/pkg/inject/inject.go` you could translate it to seconds.
As for the Helm value `WaitBeforeExitSeconds` I would keep it as is, following the convention of manifest properties like `initialDelaySeconds`, and also because at the Helm level I believe there's no function to translate a `duration` type into an int.
Let me know if this makes sense to you.